### ISSUE TITLE
Quaternion to_euler returns different values depending on angle_addition label:algebras

### ISSUE THREAD

**Issue by Gzah92** [sympy/sympy#25927]
In a toy example using a quaternion to represent a frame rotating around the x-axis, Quaternion.to_euler('zyx') returns different results depending on the setting angle_addition. As a reference the result of rotating the vector [0,0,1] is plotted as well:

![image](https://github.com/sympy/sympy/assets/10725395/39af0606-fcf5-454f-9f8e-debd5aba44e4)

The angle range of the result seems to be incorrect for the angle_addition=True case as well.

Thanks for looking into this, I really appreciate the library!

**Comment by smichr** [comment]
In the reference, just before 3.1.4, it says "It is worth noting that [Eq 32](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0276302#pone.0276302.e046) requires fewer operations than [Eq 30](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0276302#pone.0276302.e044): only 2 calls to atan2, one addition and one subtraction, but a final wrapping step may be required in order to either keep the angles either in (−π, π] or [0, 2π)." It looks to me like that wrapping step is missing.

**Comment by evbernardes** [comment]
> It looks to me like that wrapping step is missing.

Correct, adding angles always add the possibility of it falling out of one of the "canonical" ranges. I'd argue that setting `angle_addition=True` is more useful when working with symbolic variables, as the final formula is much simpler.

**Comment by merintheres** [comment]
Hi @smichr , I’d like to work on this issue. From what I understand, when angle_addition=True, the computed angles (angles0, angles2) are missing a wrapping step to keep them in the canonical range (−π, π] or [0, 2π). I’m thinking of adding this wrapping only for numeric values (leaving symbolic formulas untouched). Please let me know if I can proceed.
